### ISSUE TITLE
GroundItem.dropped not visible, but GroundItem.create are

### ISSUE THREAD

**Issue by tlf30** [apollo-rsps/apollo#352]
Hello,
So I was doing some tests. 
Performing
```
val g = GroundItem.dropped(...)
player.world.spawn(g)
```
Does not show anything in the world
But doing 
```
val g = GroundItem.create(...)
player.world.spawn(g)
```
Does show the items in the world.

I think this is an issue, or are we only supposed to run GroundItem.create when we want an item to be visible?

**Comment by garyttierney** [comment]
@Major- we discussed this on IRC before and the conclusion was that it is intended behaviour.  Is that right?  I think that needs documented.

**Comment by tlf30** [comment]
If this is intended functionality, what would be a use case for GroundItem.dropped? When a user drops an item, should that use GroundItem.create? I guess I am confused because of what the javadocs state. 
https://github.com/apollo-rsps/apollo/blob/ee78d2cd05b09782de90446a933bc1752438dbc1/game/src/main/org/apollo/game/model/entity/GroundItem.java#L19-L43

The only difference that I can tell between the two is that dropped assigns an owner. Should the item be visible to the owner when dropped?

**Comment by garyttierney** [comment]
<Major_RS> first one (create): send only to the player who dropped it at the time they drop it. second one (dropped): send to everyone, the client of the player who dropped it filters it out so it doesnt show twice

**Comment by Major-** [comment]
Oh yeah nvm this is intended, the method names aren't very useful though. Maybe we should move `GroundItem#dropped` to `Player#drop`?

**Comment by tlf30** [comment]
Understood, so when an item is dropped, then both need to get called.

**Comment by Major-** [comment]
`GroundItem.dropped` is for ground items that should only be visible to a specific player when first spawned (like when a player drops an item, or kills a monster that drops items). In these situations the dropped item is only visible to the owner (player) for 60 seconds; after that it becomes visible to everyone else. 

Use `GroundItem.create` if you want an item that is globally visible from the moment it's spawned, like for items that respawn endlessly (e.g. the mind rune in lumbridge castle), or balloons in the party room.

**Comment by tlf30** [comment]
OK, That makes sense. Thank you!

**Comment by tlf30** [comment]
So this is the test I wrote
```
import org.apollo.game.action.Action
import org.apollo.game.message.impl.InventoryItemMessage
import org.apollo.game.model.Item
import org.apollo.game.model.entity.Entity
import org.apollo.game.model.entity.EntityType
import org.apollo.game.model.entity.GroundItem
import org.apollo.game.model.entity.Player

class DropItemAction(val player: Player, val item: Int): Action<Player>(0, true, player) {
    override fun execute() {
        val region = player.world.regionRepository.fromPosition(player.position)
        if (region.getEntities<Entity>(player.position, EntityType.DYNAMIC_OBJECT, EntityType.STATIC_OBJECT).isEmpty()) {
            val amount = player.inventory.getAmount(item)
            System.out.println(amount)
            player.inventory.remove(item, amount)
            val groundItem = GroundItem.dropped(player.world, player.position, Item(item, amount), player)
            player.world.spawn(groundItem)
        } else {
            player.sendMessage("You cannot drop this here.")
        }
        stop()
    }

}

on {InventoryItemMessage::class}
        .where {option == 5 && interfaceId == 3214}
        .then {
            it.startAction(DropItemAction(it, id))
            terminate()
        }
```
When I drop an item I cannot see it, but other clients can immediately. After 60 seconds there is still no change. 

If I change it to use GroundItem.create then the item shows on all clients immediately.

**Comment by Major-** [comment]
Yeah looks like nobody ever got round to replacing the private dropped items with public ones

**Comment by tlf30** [comment]
OK, shouldn't private items at least be visible to the client that dropped them though?

**Comment by Major-** [comment]
So we need some sort of management for dropping/spawning grounditems basically

**Comment by tlf30** [comment]
No worries!

**Comment by Major-** [comment]
Blocking #349
